### ISSUE TITLE
Asynchronous smart remove no longer effective?

### ISSUE THREAD

**Issue by mauromol** [bit-team/backintime#843]
I'm running Back In Time 1.1.24 and I have the asynchronous smart remove option enabled for my backup to a remote SSH target.
However, BIT is currently removing old snapshots while running on the client side (I have the BIT icon in the system tray, with it's tooltip saying it's removing old snapshots).
On the target box, "ps aux | grep backin" reveals a single find command (to chmod u+wx), followed by "rm -rf" after the first command finishes.

I have been using the asynch removal with success for a long time and I remember perfectly that it was causing some sort of "queuing" of find/rm -rf commands on the target box without keeping BIT running on my client system.

So, was this option reverted? However I still see it in my profile options and it's enabled.

**Comment by Germar** [comment]
There was no changes in code on this. Please run `backintime --debug backup` from Terminal and post the output

**Comment by mauromol** [comment]
That's very strange. Today it happened again with the scheduled task. Then I interrupted backintime process (the tray icon was still saying it was removing old snapshots) and restarted the system, then ran from the terminal `backintime --debug --profile-id=3 backup-job` (the profile for which I see this problem is number 3) and then the async deletion worked correctly.

Could the problem be that the command line to send to the SSH server is too long and then BIT reverts to synchronous removal in this case? If so, I could explain this by saying that the scheduled job had already removed some snapshots and then, when I ran BIT from the command line, the remaining ones were producing a shorter command line, hence it succeeded...

**Comment by Germar** [comment]
BiT check the max argument lenght for command on the SSH server before running the command and will splitt the command into multiple sessions if the would be to long.

And even if this would have been the case, as soon as you ran with `--debug` the command was even longer because of debug messages printed on the remote server.

Please run `backintime-qt4 --debug` and open/close Settings once. This should add `--debug` to the cronjobs, too. (Please double check with `crontab -l`). Don't start BiT again without `--debug` as this would rewrite the cronjob without debug. Next time you stumbling on this please post the part from your syslog.

**Comment by mauromol** [comment]
Hi Germar, I was not able to reproduce again this problem, although I'm sure it did happen for a while.

Anyway, what I'm observing now is that the smart remove in background feature is failing and I find this in my syslog:

    WARNING: Command "ssh ... [very very long command] 9>\"/path/to/backup/smartremove.lck\""'" returns 32512

Any idea of what can be causing that "returns 32512"?

**Comment by emtiu** [comment]
Hi, we're sorting out old issues, so could you tell us: Is this problem still relevant?

**Comment by mauromol** [comment]
I've stopped using BIT a while ago in favor of another solution (borg), at least on my main PC. I use it on other systems.
Yet, I can't remember the hereby described problem to occur during my latest usages.
What was surely relevant in my latest usages on my main PC was #1027, for which I had to manually patch BIT sources, as described in that report.

I'm glad to know someone is picking up BIT again, I think it's still a great software and still precious on some relevant use cases.
